.github/workflows/go.yml

```diff
@@ -57,6 +59,25 @@ jobs:
       with:
         version: v1.51.2
 
+  coverage:
+    name: coverage
+    permissions:
```

Comment:
```suggestion
    permissions:
      contents: read
```

Reply:
Fwiw, this does a couple of things and should be included in `build:` as well...

specifically, if you don't specify any permissions, each workflow/job will default to some level of maximal permissions depending on various things. By specifying _any_ permissions, the permissions categories you don't specify get no permission (metadata being a caveat, but...) -- this is a good thing -- you're limiting permissions to only those you ask for, instead of unintentionally allowing potentially quite generous permissions.